src/ConstraintSolver.jl

```diff
@@ -56,29 +56,33 @@ struct NoObjective <: ObjectiveFunction end
 
 mutable struct BasicConstraint <: Constraint
     idx                 :: Int
-    fct                 :: Function
+    fct                 :: Union{MOI.AbstractScalarFunction, MOI.AbstractVectorFunction}
+    set                 :: Union{MOI.AbstractScalarSet, MOI.AbstractVectorSet}
     indices             :: Vector{Int}
     pvals               :: Vector{Int}
     hash                :: UInt64
-    BasicConstraint() = new()
 end
 
 # support for a <= b constraint
 mutable struct SingleVariableConstraint <: Constraint
     idx                 :: Int
-    fct                 :: Function
+    fct                 :: MOI.AbstractScalarFunction
+    set                 :: MOI.AbstractScalarSet
     indices             :: Vector{Int}
     pvals               :: Vector{Int}
     lhs                 :: Int
     rhs                 :: Int
     hash                :: UInt64
-    SingleVariableConstraint() = new()
 end
 
 struct AllDifferentSet <: MOI.AbstractVectorSet
     dimension :: Int64
 end
 
+struct EqualSet <: MOI.AbstractVectorSet
```

Comment:
EqualSet is just taking the dimensions but NotEqual takes a value?

Reply:
`EqualSet` and `AllDifferentSet` are used in this way:
`vars in Set` but `NotEqualSet` is used for `!=` which means it is a scalar.